service/history/queryRegistry.go

```diff
@@ -40,14 +40,16 @@ type (
 		getCompletedIDs() []string
 		hasUnblockedQuery() bool
 		getUnblockedIDs() []string
+		hasFailedQuery() bool
```

Comment:
There are quite a few methods in this interface that are for testing purpose only. Can we remove them and only leave the "real" ones there?
For testing you can use the assertion or test against the impl directly.

Reply:
I don't feel that strongly but I did think about this and I prefer to leave it like this for the following reasons

- This is a registry API, so even though many of these APIs are not used they are reasonable methods to have on this primitive. 
- I find it confusing to read the interface if you can get completed queries and buffered queries only and yet the user needs to know that queries can also be failed, or unblocked.